CONTRIBUTING.md

```diff
@@ -1,7 +1,186 @@
+- applied the [code formatter](#code-style-and-formatting);
+- [documented](#code-style-and-formatting) your code;
+- written [tests](#writing-tests) that cover your code; and
+- accompanied any remaining `TODO`s or `FIXME`s with a link to an active [issue](#reporting-issues).
+
+### Feature branches
+Develop new changes in a feature branch or in a fork (if you have no write permission on the repository). Please use an informative branch name and use kebab case (e.g., `my-new-feature`).
+
+### Commit messages
+We currently do not adhere to strict rules regarding commit messages. We only ask that your commit messages are descriptive and informative.
+
+### Pull requests
+When you file a PR, provide a clear and well-written description of the changes you are proposing. We use PRs to automatically construct our changelog and release notes; the text you write will be featured in them.
+
+#### PR titles
+Please make sure that the title of your PR adheres to the following rules:
+1. Describe a contribution, not the activity of implementing it.
+  - **Bad**: "Improve reporting of file system errors"
+  - **Good**: "Improved reporting of file system errors"
+  - **Bad**: "Fixes #123"
+  - **Good**: "Fix for #123"
+  - **Bad**: "Optimizing algorithm x"
+  - **Good**: "Optimization of algorithm x"
+2. Start with a capital.
+  - **Bad**: "in a hurry"
+  - **Good**: "Hurried new feature"
+3. Do not use title case.
+  - **Bad**: "Amazing Fix of a Terrible Bug"
+  - **Good**: "Amazing fix of a terrible bug"
+4. Do not use a period at the end.
+  - **Bad**: "Titles are not sentences."
+  - **Good**: "Titles are not sentences"
+
+#### Labeling PRs
+Labels are used to organize our changelog and release notes. Please label your PR to make it as clear as possible from the labels what this PR is about. If, for whatever reason, your changes should not appear in these digests, you can mark it as such using the `exclude` label.
+
+#### Draft PRs
+If a PR is not yet ready for review, mark it as **Draft**. Once it is ready for review, mark it **Ready for Review**.
+
+#### Merging
+Perform merges to bring your feature branch up-to-date with master locally (do not issue a PR). This is very easy to do.
+1. Ensure you are on your feature branch (run `git branch` to find out; `git switch my-feature-branch` to switch to your feature branch).
+2. Make sure you have the latest changes (run `git fetch --all`)
+3. Perform the merge (run `git merge origin/master`, assuming that you want to merge the branch `master` from remote `origin` into `your-feature-branch`).
+
+#### Addressing reviews
+To address feedback from code review, implement changes and push to your feature branch. If you are certain that a reviewer's concern has been addressed by your new changes, hit the `Resolve conversation` button. If you are unsure, ask for follow up in the conversation and let the reviewer determine whether the feedback was addressed accordingly.
+
+### Code style and formatting
```

Comment:
We should mention and/or link our own guidelines here.

Reply:
They are all mentioned in this section. There is no separate `code-style.md` anymore.